idx/memory/memory.go

```diff
@@ -639,42 +597,75 @@ func (m *MemoryIdx) Prune(orgId int, oldest time.Time) ([]idx.Archive, error) {
 	return pruned, nil
 }
 
-// filepath.Match doesn't support {} because that's not posix, it's a bashism
-// the easiest way of implementing this extra feature is just expanding single queries
-// that contain these queries into multiple queries, who will be checked separately
-// and whose results will be ORed.
-func expandQueries(query string) []string {
-	queries := []string{query}
+func getMatcher(path string) (func([]string) []string, error) {
+	// Matches everything
+	if path == "*" {
+		return func(children []string) []string {
+			log.Debug("memory-idx: Matching all children")
+			return children
+		}, nil
+	}
+
+	// Convert to regex and match
+	if strings.ContainsAny(path, "*{}[]?") {
```

Comment:
That's fair. This was mostly to simplify the search code to run a single iteration through each node. I could instead make this optimization by having the exact match closure operate on multiple strings.

The logic would then be:
If there are special characters that are not {}, form the regex (which may include conversion of {} to regex)
else match 1 or more exact.